server/routes/probationPractitionerReferrals/interventionProgressPresenter.ts

```diff
@@ -39,30 +41,46 @@ export default class InterventionProgressPresenter {
     }
 
     return this.actionPlanAppointments.map(appointment => {
+      const sessionTableParams = this.sessionTableParams(appointment)
+
       return {
         sessionNumber: appointment.sessionNumber,
         appointmentTime: DateUtils.formatDateTimeOrEmptyString(appointment.appointmentTime),
-        tagArgs: this.tagArgs(appointment),
-        linkHtml: appointment.sessionFeedback?.submitted ? `<a class="govuk-link" href="#">View</a>` : '',
+        tagArgs: { text: sessionTableParams.text, classes: sessionTableParams.tagClass },
+        linkHtml: sessionTableParams.linkHTML,
       }
     })
   }
 
-  private tagArgs(appointment: ActionPlanAppointment): Record<string, unknown> {
-    const sessionFeedbackAttendance = appointment.sessionFeedback?.attendance
-
-    if (sessionFeedbackAttendance?.attended === 'no') {
-      return { text: 'FAILURE TO ATTEND', classes: 'govuk-tag--purple' }
-    }
-
-    if (sessionFeedbackAttendance?.attended === 'yes' || sessionFeedbackAttendance?.attended === 'late') {
-      return { text: 'COMPLETED', classes: 'govuk-tag--green' }
-    }
+  private sessionTableParams(appointment: ActionPlanAppointment): { text: string; tagClass: string; linkHTML: string } {
+    const status = sessionStatus.forAppointment(appointment)
+    const presenter = new SessionStatusPresenter(status)
 
-    if (appointment.appointmentTime) {
-      return { text: 'SCHEDULED', classes: 'govuk-tag--blue' }
+    switch (status) {
+      case SessionStatus.didNotAttend:
+        return {
+          text: presenter.text,
+          tagClass: presenter.tagClass,
+          linkHTML: `<a class="govuk-link" href="#">View feedback form</a>`,
```

Comment:
> This isn't very DRY (there's a lot we can re-use from the SP version of
this) but until we know what the links we're going to be using look
like, I'm keeping them separate.

Looks like this bit of the commit message is no longer applicable?